src/main/java/core/db/TransferTypeTable.java

```diff
@@ -19,23 +19,17 @@ protected void initColumns() {
 		columns[1]= new ColumnDescriptor("TYPE",Types.INTEGER,true);
 
 	}
-	
-    void setTransferType(int playerId, int type) {
-        final String query = "update " + getTableName() + " set TYPE = " + type + " where PLAYER_ID ="
-                       + playerId; 
-        final int count = adapter.executeUpdate(query);
 
-        if (count == 0) {
-        	adapter.executeUpdate("insert into " + getTableName()
-                                                          + " (PLAYER_ID, TYPE) values ("
-                                                          + playerId + "," + type + ")"); 
-        }
+    void setTransferType(int playerId, int type) {
+        executePreparedDelete(playerId);
+        executePreparedInsert(
+                playerId,
+                type
+        );
```

Comment:
You seem to have gone for this pattern in multiple places of first deleting the existing entry of same ID, and then inserting the new one, rather than updating – which means that this operation is no longer atomic.  This could potentially lead to some troubles.  Any concern there?

Reply:
the advantage i see for this pattern is, that it works in both cases: initial creation of the record and its update. 
checking if records exists and doing either an update or an insert depending on the result sounds more complicated - but maybe, this is a mistake?!